### PR TITLE
schedulers: enable placement rules in tests

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Close https://github.com/tikv/pd/issues/5956

### What is changed and how does it work?

1. enable placement rules in tests

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)

### Release note

```release-note
None.
```

### CODE REVIEW

File: server/schedulers/utils.go [hunk 1]
```diff
@@ -375,3 +379,18 @@ func (q *retryQuota) GC(keepStores []*core.StoreInfo) {
 		}
 	}
 }
+
+func newTestCluster(needToRunStream ...bool) (context.CancelFunc, *config.PersistOptions, *mockcluster.Cluster, *schedule.OperatorController) {
```
Comment (rleungx) [inline server/schedulers/utils.go:383]: Shall we move it to testutil?

Comment (lhy1024) [reply server/schedulers/utils.go:383]: This function will remain in the schedulers pkg, as it depends on schedulers.register

**Comment by lhy1024** [comment]
cc @rleungx @nolouch PTAL

**Review by rleungx — APPROVED** [review]

**Review by nolouch — APPROVED** [review]

**Pull request merged**